broker/get-policy.go

```diff
@@ -19,9 +21,32 @@ func GetPolicyFx(w http.ResponseWriter, r *http.Request) (string, interface{}, e
 	policyUid := chi.URLParam(r, "uid")
 	log.Printf("policyUid: %s", policyUid)
 
-	firePolicy := lib.GetDatasetByEnv(r.Header.Get("Origin"), lib.PolicyCollection)
+	idToken := r.Header.Get("Authorization")
+	authToken, err := lib.GetAuthTokenFromIdToken(idToken)
+	if err != nil {
+		log.Printf("error fetching authToken: %s", err.Error())
+		return "", nil, err
+	}
+
+	policy, err := plc.GetPolicy(policyUid, lib.PolicyCollection)
+	if err != nil {
+		log.Printf("error fetching policy: %s", err.Error())
+		return "", nil, err
+	}
+
+	switch authToken.Role {
+	case lib.UserRoleCustomer:
+		if policy.Contractor.Uid != authToken.UserID {
+			log.Printf("error fetching policy invalid user id: %s", authToken.UserID)
+			return "", nil, errors.New("invalid user id")
+		}
+	case lib.UserRoleAreaManager, lib.UserRoleAgency, lib.UserRoleAgent:
+		if !utils.CanBeAccessedBy(authToken.Role, policy.ProducerUid, authToken.UserID) {
+			log.Printf("error fetching policy invalid producer uid: %s", authToken.UserID)
+			return "", nil, errors.New("invalid producer uid")
+		}
+	}
```

Comment:
remove `case lib.UserRoleCustomer` since we don't have yet an authenticated flow for e-commerce, and replace it with a single `if` statement.